tests/Adapter/TolerantParser/ClassToFile/Transformer/FixNamespaceTransformerTest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Phpactor\CodeTransform\Tests\Adapter\TolerantParser\ClassToFile\Transformer;
+
+use Phpactor\ClassFileConverter\Adapter\Composer\ComposerFileToClass;
+use Phpactor\ClassFileConverter\Adapter\Simple\SimpleFileToClass;
+use Phpactor\ClassFileConverter\Domain\ClassToFile;
+use Phpactor\ClassFileConverter\Domain\FileToClass;
+use Phpactor\CodeTransform\Adapter\TolerantParser\ClassToFile\Transformer\FixNameTransformer;
+use Phpactor\CodeTransform\Domain\SourceCode;
+use Phpactor\CodeTransform\Tests\Adapter\AdapterTestCase;
+
+class FixNamespaceTransformerTest extends AdapterTestCase
+{
+    /**
+     * @dataProvider provideFixClassName
+     */
+    public function testFixClassName(string $filePath, string $test)
+    {
+        $workspace = $this->workspace();
+        $workspace->reset();
+        $workspace->loadManifest(file_get_contents(__DIR__ . '/fixtures/' . $test));
+        $source = $workspace->getContents($filePath);
+        $expected = $workspace->getContents('expected');
+
+        $fileToClass = new SimpleFileToClass($this->workspace()->path('/'));
```

Comment:
Note that `SimpleFileToClass` doesn't predict class names, it scans the file and returns the _actual_ class, so not suitable here (should use composer one instead).